pkg/api/v1beta3/types.go

```diff
@@ -1051,6 +1051,8 @@ type StatusDetails struct {
 	// The Causes array includes more details associated with the StatusReason
 	// failure. Not all StatusReasons may provide detailed causes.
 	Causes []StatusCause `json:"causes,omitempty" description:"the Causes array includes more details associated with the StatusReason failure; not all StatusReasons may provide detailed causes"`
+	// If specified, the time in seconds before the operation should be retried.
+	RetryAfter int `json:"retryAfter,omitempty" description:"the number of seconds before the client should attempt to retry this operation"`
```

Comment:
I think this would be read from the header field and set into the returned object.  The header has higher priority, like the http status code.  It's mostly a convenience to pass this back to client code that may not want/need to see http specific info.